examples/Services/Planning/AddKeywordPlan.cs

```diff
@@ -0,0 +1,322 @@
+            {
+                string keywordPlanResource = CreateKeywordPlan(client, customerId);
+                string planCampaignResource = CreateKeywordPlanCampaign(client, customerId,
+                    keywordPlanResource);
+                string planAdGroupResource = CreateKeywordPlanAdGroup(client, customerId,
+                    planCampaignResource);
+                CreateKeywordPlanKeywords(client, customerId, planAdGroupResource);
+                CreateKeywordPlanNegativeKeywords(client, customerId, planCampaignResource);
+            }
+            catch (GoogleAdsException e)
+            {
+                Console.WriteLine("Failure:");
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+            }
+        }
+
+        /// <summary>
+        /// Creates the keyword plan.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <returns>The newly created keyword plan resource.</returns>
+        private string CreateKeywordPlan(GoogleAdsClient client, long customerId)
+        {
+            // Get the KeywordPlanService.
+            KeywordPlanServiceClient serviceClient = client.GetService(
+                Services.V0.KeywordPlanService);
+
+            // Create a keyword plan for next quarter forecast.
+            KeywordPlan keywordPlan = new KeywordPlan()
+            {
+                Name = "Keyword plan for traffic estimate #" + ExampleUtilities.GetRandomString(),
+                ForecastPeriod = new KeywordPlanForecastPeriod()
+                {
+                    DateInterval = KeywordPlanForecastInterval.NextQuarter
+                }
+            };
+
+            KeywordPlanOperation operation = new KeywordPlanOperation()
+            {
+                Create = keywordPlan
+            };
+
+            // Add the keyword plan.
+            MutateKeywordPlansResponse response = serviceClient.MutateKeywordPlans(
+                customerId.ToString(), new KeywordPlanOperation[] { operation });
+
+            // Display the results.
+            String planResource = response.Results[0].ResourceName;
+            Console.WriteLine($"Created keyword plan: {planResource}.");
+            return planResource;
+        }
+
+        /// <summary>
+        /// Creates the campaign for the keyword plan.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="keywordPlanResource">The keyword plan resource.</param>
+        /// <returns>The newly created campaign resource.</returns>
+        private string CreateKeywordPlanCampaign(GoogleAdsClient client, long customerId,
```

Comment:
Is Campaign a necessary resource for keyword planner to work?  If not, it may be clearer for the comment to reflect that as an optional input.

Reply:
Yes, it is a required parameter. The UI hides these steps from an end user.